master/freebsd_ci.py

```diff
@@ -0,0 +1,227 @@
+    ),
+
+    steps.GitHub(
+        repourl='git://github.com/JuliaLang/julia.git',
+        mode='full',
+        method='clean'),
+
+    steps.ShellCommand(
+        name='cleanall',
+        command=['gmake', 'cleanall'],
+    ),
+
+    steps.ShellCommand(
+        name='cleanup',
+        command=['./.freebsdci.sh', 'cleanup']
+    ),
+
+    steps.Compile(
+        command=['./.freebsdci.sh', 'compile'],
+        env={
+            'WORKERNAME': util.Property('workername'),
+            'MAKE_JOBS_NUMBER': util.Property('make_jobs'),
+        },
+    ),
+
+    steps.ShellCommand(
+        name='build-stats',
+        command=['./.freebsdci.sh', 'build-state'],
+    ),
+
+    steps.Test(
+        command=['./.freebsdci.sh', 'runtests'],
+        env={
+            'WORKERNAME': util.Property('workername'),
+            'MAKE_JOBS_NUMBER': util.Property('make_jobs'),
+        },
+        timeout=1200,  # 20 min
+    ),
+
+    steps.ShellCommand(
+        name='test embedding',
+        command=['./.freebsdci.sh', 'test-embedding'],
+        env={
+            'WORKERNAME': util.Property('workername'),
+            'MAKE_JOBS_NUMBER': util.Property('make_jobs'),
+        },
+    ),
+])
+
+freebsdci_test_factory = util.BuildFactory()
+freebsdci_test_factory.addSteps([
+    steps.BSDSysInfo(),
+    steps.BSDSetMakeVar(['make_jobs'], ['MAKE_JOBS_NUMBER']),
+
+    steps.ShellCommand(
+        name='cleanup stdlib',
+        command=['rm', '-rvf', 'stdlib'],
+    ),
+
+    steps.ShellCommand(
+        name='testing var',
+        command=['/bin/sh', '-c' , 'echo $CHANGES'],
+        env={
+            'CHANGES': util.Property('branch'),
+            'MYVAR': 'foo',
+            'WORKERNAME': util.Property('workername'),
+        }
+    ),
+
+    steps.GitHub(
+        repourl='git://github.com/JuliaLang/julia.git',
+        mode='full',
+        method='clean'),
+
+    steps.ShellCommand(
+        name='cleanall',
+        command=['gmake', 'cleanall'],
+    ),
+
+    steps.ShellCommand(
+        name='cleanup',
+        command=['./.freebsdci.sh', 'cleanup']
+    ),
+
+    steps.Compile(
+        command=['./.freebsdci.sh', 'compile'],
+        env={
+            'WORKERNAME': util.Property('workername'),
+            'MAKE_JOBS_NUMBER': util.Property('make_jobs'),
+        },
+    ),
+
+    steps.ShellCommand(
+        name='build-stats',
+        command=['./.freebsdci.sh', 'build-state'],
+    ),
+
+    steps.Test(
+        command=['./.freebsdci.sh', 'runtests'],
+        env={
+            'WORKERNAME': util.Property('workername'),
+            'MAKE_JOBS_NUMBER': util.Property('make_jobs'),
+        },
+        timeout=1200,  # 20 min
+    ),
+
+    steps.ShellCommand(
+        name='test embedding',
+        command=['./.freebsdci.sh', 'test-embedding'],
+        env={
+            'WORKERNAME': util.Property('workername'),
+            'MAKE_JOBS_NUMBER': util.Property('make_jobs'),
+        },
+    ),
+])
```

Comment:
Long-term, I would like the FreeBSD config to mirror the rest of the world more exactly. E.g. just have this be another job that is run by the factory in package.py

Reply:
well, I don't get your point. Could you explain more?

Reply:
I think the idea is that this has a lot in common wit the structure in https://github.com/JuliaCI/julia-buildbot/blob/master/master/package.py. So the wuestion is can we factor out the FreeBSD specific steps and reuse `package.py`

Reply:
Yes, what I want to do is have only the `package.py` file contain the code for actually building on all platforms, with `doStepIf=isbsd` or something like that for the BSD-specific setup.